ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -353,21 +394,36 @@ public boolean equals(Object o) {
 
   @Override
   public int hashCode() {
-    return Utils.hashcode(new Object[]{version, partitionId, uuid});
+    return Utils.hashcode(new Object[]{uuid});
   }
 
-  /**<p>
-   *   Gets the value of {@link #CURRENT_VERSION}.
-   * </p>
-   * <p>
-   *   It is typically not a good practice to call overridable methods from constructor. However, there could not be
-   *   cleaner way to set the {@link #version} field by either calling this method, or by reading from an input stream
-   *   from {@link BlobId#BlobId(String, ClusterMap)}, and also because this method simply returns a static final value
-   *   that does not depend on the state of an instance.
-   * </p>
-   * @return The value of {@link #CURRENT_VERSION}.
+  /**
+   * @return The version of this BlobId
    */
-  protected short getCurrentVersion() {
-    return CURRENT_VERSION;
+  protected short getVersion() {
+    return version;
+  }
+
+  /**
+   * @return all valid versions of BlobId.
+   */
+  public static Short[] getAllValidVersions() {
```

Comment:
decision b/w `short` and `Short`? I prefer `short` here since this cannot be `null`

Reply:
Also does this need to be `public`?

Reply:
Seems like it's only used in tests now - do you envision a production use case for this?